cps/src/main/js/workflow-editor.js

```diff
@@ -48,7 +48,34 @@ jenkinsJSModules.import('ace-editor:ace-editor-122')
                 });
 
                 editor.on('blur', function() {
-                    textarea.trigger('change');
```

Comment:
Obviously you would delete the Groovy compilation from `doCheckScript`.

Approval warnings just look weird as editor annotations, when it is not about a specific line.

You could use one AJAX method if you dug up the code to display `FormValidation` from `hudson-behavior.js`.